lib/srv/discovery/discovery.go

```diff
@@ -554,6 +580,86 @@ func (s *Server) Wait() error {
 	return nil
 }
 
+func (s *Server) watchCARotations(ctx context.Context) {
```

Comment:
We have a very similar logic in [lib/service/connect.go:(*TeleportProcess).syncRotationStateCycle](https://github.com/gravitational/teleport/blob/master/lib/service/connect.go#L726). Is there any way to reuse it or hook into it to avoid duplicating the same watcher? Maybe parameterize it with the types of CAs we want to watch and have it emit some hook/event we can subscribe to here in the discovery service?

@espadolini What do you think, would that make sense?

Reply:
This watcher is different enough that I don't think it would be worth it to unify them (here we only act on certain phase changes, for instance).

A bigger problem is that until we figure out how to do reloadless CA rotations, this nice watcher loop will just get closed whenever the host CA is rotated (other than the`init` and `rollback` phases, at least) - as it is, it's kinda pointless to watch for host CA rotations of the local CA, since we're just going to be closing the `TeleportProcess` at some unspecified time in the future, so you don't really have guaranteed time to do anything.